src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -146,6 +146,19 @@ def create_role_assignment(cmd, role, assignee=None, assignee_object_id=None, re
     if assignee_principal_type and not assignee_object_id:
         raise CLIError('usage error: --assignee-object-id GUID [--assignee-principal-type]')
 
+    # If assignee object id is provided without assignee principal type, try to get principle type from graph
+    if assignee_object_id and not assignee_principal_type:
```

Comment:
Per my understanding, the requirement of #16006 and email "Azure RBAC - Support passing attribute principalType during PUT operations HTTP calls" is to pass `principalType` whenever possible, not only when `--assignee-object-id` is provided.

In other words, this implementation passes `principalType` only when `--assignee-object-id` is provided. If `--assignee` is provided (user, group, service principal), `principalType` will not be passed.

Besides, the value assigned in `--assignee` will be converted to object ID anyway:

https://github.com/Azure/azure-cli/blob/5544a0f3bc32aa778d296f71c4d37e48f8ebe691/src/azure-cli/azure/cli/command_modules/role/custom.py#L1783

Reply:
Resolved by
replacing `_resolve_object_id` in `_create_role_assignment`
with `_resolve_assignee_object` in `create_role_assignment`.

The latter will process differently between `assignee` and `assignee_object_id` and return both `object_id` and `object_type`